test/e2e/specs/fakerp/keyrotation.go

```diff
@@ -64,12 +64,35 @@ var _ = Describe("Key Rotation E2E tests [KeyRotation][Fake][LongRunning]", func
 		Expect(internalAfterUpdate).NotTo(BeNil())
 
 		By("Verifying that the initial config blob does not match the one created after the update")
-		configMatch := reflect.DeepEqual(internal.Config.Certificates, internalAfterUpdate.Config.Certificates)
+		configMatch := reflect.DeepEqual(internal.Config.Certificates, updated.Config.Certificates)
 		Expect(configMatch).To(BeFalse())
 
-		By("Verifying that the mutated config blob matches the one created after the update")
-		configMatch = reflect.DeepEqual(mutated.Config.Certificates, internalAfterUpdate.Config.Certificates)
-		Expect(configMatch).To(BeTrue())
+		By("Verifying that the updated internal config blob and the config returned after the update contain matching ca certificates")
+		Expect(reflect.DeepEqual(internalAfterUpdate.Config.Certificates.ServiceSigningCa.Cert, updated.Config.Certificates.ServiceSigningCa.Cert)).To(BeTrue())
```

Comment:
Instead of comparing `updated` with `internalAfterUpdate`, a more honest check would be `updated` with `internal` and expect it to be false.

Reply:
Also, ensure you deep copy `internal` before mutating its state.

Reply:
Actually, the ideal test would minimize the amount of steps that deal with the internal config blob with the goal being to totally remove dealing with the internal config blob. Ideally, as an client-side admin test, you should GET the OSA resource, then do the key rotation RPC call, then compare the response with the result from GET.

Reply:
This can be improved in the future, I think there might be some work on making GET foolproof in the fake rp.

/lgtm

Reply:
> This can be improved in the future, I think there might be some work on making GET foolproof in the fake rp.

Opened https://github.com/openshift/openshift-azure/pull/985